earthfile2llb/validator.go

```diff
@@ -0,0 +1,55 @@
+package earthfile2llb
```

Comment:
Could this file go under the `ast` package and have public functions?

Reply:
Yeah; I had started here hoping to do a little more but decided it wasnt worth it. Ill move it to finish formalizing that